src/sql/workbench/contrib/commandLine/electron-sandbox/commandLine.ts

```diff
@@ -171,53 +171,71 @@ export class CommandLineWorkbenchContribution implements IWorkbenchContribution,
 		}
 	}
 
+	/**
+	 * Handles user provided URL to initiate command handler for supported commands.
+	 * @param uri User provided URL in format: azuredatastudio://{command}?{option1}={value1}&{option2}={value2}...
+	 * @returns True if URL was opened successfully, false otherwise
+	 * @throws Error if invalid URL handler used.
+	 */
 	public async handleURL(uri: URI): Promise<boolean> {
 		let key = uri.authority;
+		let args = this.parseProtocolArgs(uri);
 
+		const result = await this.runCommandHandler(key, args);
+		if (typeof result !== 'boolean') {
+			throw new Error('Invalid URL Handler used in commandLine code.');
+		}
+
+		return result;
+	}
+
+	private async runCommandHandler(key: string, args: NativeParsedArgs) {
 		let method = pathMappings[key];
 
 		if (!method) {
 			return false;
 		}
 		method = method.bind(this);
-		const result = await method(uri);
-
-		if (typeof result !== 'boolean') {
-			throw new Error('Invalid URL Handler used in commandLine code.');
-		}
-
-		return result;
+		return await method(args);
 	}
 
 	@pathHandler({
-		path: 'connect'
+		path: Command.connect
 	})
-	public async handleConnect(uri: URI): Promise<boolean> {
+	public async handleConnect(args: NativeParsedArgs): Promise<boolean> {
 		try {
-			let args = this.parseProtocolArgs(uri);
 			if (!args.server) {
-				this._notificationService.warn(localize('warnServerRequired', "Cannot connect as no server information was provided"));
+				this._notificationService?.warn(localize('warnServerRequired', "Cannot connect as no server information was provided"));
```

Comment:
We should never have an undefined notification service...if this is for tests then I would suggest using this : https://github.com/Microsoft/azuredatastudio/blob/main/src/vs/platform/notification/test/common/testNotificationService.ts#L10

Reply:
(same for all the other places you do that in this file)